packages/components/src/components/TokenGate/TokenGate.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState, ReactNode } from 'react';
+import { ethers } from 'ethers';
+import { Spinner } from '@chakra-ui/react';
+export interface TokenGateProps {
+  /**
+   * The provider or signer to fetch the address from the ens
+   */
+  provider: ethers.providers.Web3Provider;
+  /**
+   * The token contract address
+   */
+  contractAddress: string;
+  /**
+   * The token quantity required to access child component. Default=1
+   */
+  requiredQuantity?: number;
+  /**
+   * Child nodes
+   */
+  children: ReactNode;
+  /**
+   * Optional message if access denied
+   */
+  deniedMessage?: ReactNode;
+  loader?: boolean;
+}
+export const TokenGate: React.FC<TokenGateProps> = ({
+  contractAddress,
+  provider,
+  requiredQuantity = 1,
+  children,
+  deniedMessage,
+  loader = false,
+}) => {
+  const erc20Abi = ['function balanceOf(address owner) view returns (uint256)'];
```

Comment:
you can import this from the constants file of this library 
import {ERC20ABI} from '@web3-ui/hooks';

Reply:
Maybe i'll have to merge the main into my branch first cause i get an error that it doesn't exist.